rewrite-core/src/main/java/org/openrewrite/config/DeclarativeRecipe.java

```diff
@@ -118,7 +120,6 @@ private void initialize(List<Recipe> uninitialized, List<Recipe> initialized, Co
                 initialized.add(recipe);
             }
         }
-        uninitialized.clear();
```

Comment:
One thing we need to consider: Currently the `uninitializedRecipes` and `uninitializedPreconditions` fields will get serialized by Jackson when serializing the recipe. Do you see any chance that this will cause problems now that we don't set the list to the empty list in the end?

Reply:
Personally I think we should be able to annotate these as `@JsonIgnore`, but I can't be sure...

Reply:
I believe that we replace the recipes with shrinkwrap versions everywhere where they are potentially serialized, otherwise, the recipeList would cause the same issue anyway, but we can add @JsonIgnore to all of them if you prefer